### DISCUSSION TITLE
Does webrtc-SCTP rely on UDP?

### DISCUSSION THREAD

**Discussion by huixue** [webrtc-rs/webrtc#380] [General]
From the dep graph here: 
https://webrtc.rs/res/webrtc_stack.png

I guess webrtc-SCTP in this implementation relies on UDP to work? However it conceptually should not be this way, where SCTP could just run on top of IP.

**Comment by huixue** [comment]
I'm asking, because I'm using webrtc/sctp/examples/pong.rs, and use another SCTP client trying to connect to the server, but because pong.rs is listening on UDP, the SCTP client couldn't connect to it.

**Reply by k0nserv** [reply]
Yes, in webrtc-rs SCTP runs on top of the ICE transport which is either UDP or TCP(in the case of TURN being used).

That said, this is abstracted away from the SCTP implementation itself via the `Conn` trait from the `webrtc-util` crate. It might be possible to implement a `Conn` directly over ip to avoid UDP and TCP if you wish
